plugins/tech-insights/src/api/TechInsightsClient.ts

```diff
@@ -45,76 +45,76 @@ export class TechInsightsClient implements TechInsightsApi {
     this.renderers = options.renderers;
   }
 
+  async getLatestFacts(
+    entity: CompoundEntityRef,
+    facts: string[],
+  ): Promise<InsightFact> {
+    const { namespace, kind, name } = entity;
+    const entityQuery = `entity=${encodeURIComponent(
+      kind.toLocaleLowerCase('en-US'),
+    )}:${encodeURIComponent(namespace)}/${encodeURIComponent(name)}`;
```

Comment:
Should probably always form queries using `URLSearchParams`. Also should stringify using the builtin functions

```ts
import { stringifyEntityRef } from '@backstage/catalog-model';

const entityQuery = new URLSearchParams({ entity: stringifyEntityRef(entity) }).toString();
```

There's also some usage in the repo of the `qs` library that could help formulate the entire query including the ids with the right syntax, if you gave it the right flags. Would be more difficult to accidentally forget some encoding and stuff that way (i see the ids aren't escaped below for example)